raiden/tests/integration/api/rest/test_rest.py

```diff
@@ -864,30 +894,47 @@ def test_payment_events_endpoints(
         },
     )
     assert len(json_response) == 1
-    # test endpoint without partner for app0 but with limit/offset to get only second
+
+    # test endpoint without partner for app0 but with limit/offset
+    # to get only second transfer of token_address
     request = grequests.get(
         api_url_for(
-            api_server_test_instance,
-            "token_paymentresource",
-            token_address=token_address,
-            limit=1,
-            offset=1,
+            app0_server, "token_paymentresource", token_address=token_address, limit=1, offset=1,
         )
     )
     response = request.send().response
     assert_proper_response(response, HTTPStatus.OK)
     json_response = get_json_response(response)
+
+    # this should return only payment 3, since payment 1 is offset
+    # and payment 2 is of another token address
+    assert len(json_response) == 1
     assert must_have_event(
         json_response,
         {
             "event": "EventPaymentSentSuccess",
-            "identifier": str(identifier2),
-            "target": to_checksum_address(target2_address),
+            "identifier": str(identifier3),
+            "token_address": to_checksum_address(token_address),
+        },
+    )
+    assert not must_have_event(
```

Comment:
Instead of `assert not`, it would be better to have every test with `assert len() == X`, this way we check for the absence of any invalid value, regardless of its content.